### ISSUE TITLE
[Feature Request] Echo nest is dead.  Long live Spotify

### ISSUE THREAD

**Issue by wagnered** [ampache/ampache#2443]
I just learned that the echo nest developer api was discontinued in 2016.  It was bought out by Spotify and they of course recommend that the Spotify api be used.

I was looking into the Spotify api because the Musicbrainz' was a bust for artist images.  What was returned was web pages with the requested image.  I was going to just use a query to retrieve artist info and thus urls to actual artist images.  I didn't intend a full access component.

I would like to suggest a full use Spotify component to replace echo nest.

**Comment by wagnered** [comment]
I have the `gather_spotify` function for both album and artist art working.
I see a logistical problem if lastFM is listed before Spotify in `art order`: lastFM will fill the artist image with the "not found"generic image and Ampache won't replace them with images from  spotify. Now looking for the best remedy.

BTW I've taken the liberty of removing echo nest from composer.json as it was shut down about 4 years ago.

**Comment by lachlan-00** [comment]
will it happen on the latest develop? i took out the artist search so i'm hoping that will return nothing and skip to the next?

**Comment by wagnered** [comment]
I just update from upstream and noticed the artist bit absent.  But ran into other trouble.  I had removed the echonest package just after updating. 

 I didn't know where to start looking for the problem.  Luckily I managed to get vim + vdebug working and was able to step through the code.  I found out that tiny.lib.php loaded the echonest module. 

Now I'm back to square one getting the 400 error code. 400 implies that the data is corrupted.  Probably a typo somewhere.

**Comment by lachlan-00** [comment]
An easy test might be to open a pull without actually opening and compare the difference?

I'm on a full day at my office but I'll do the same when I have a break.

**Comment by wagnered** [comment]
No, I added an extra space when I entered the secret in ampache.cfg.php.  it's working quite well now

I want to to add a try/catch around the logic that requests  the authorization token.

**Comment by wagnered** [comment]
BTW: Is it ok to bump the config file to 43?

**Comment by lachlan-00** [comment]
i think we should. it's just an int and it's an important update for develop people too

**Comment by wagnered** [comment]
Ok, I claim the number 43.

I was just thinking that even though the exception is being caught, it won't stop the spamming of Spotify servers: This could be sever if there are a considerable number of songs.
 The only way to stop the spamming is to restart the webserver.  I was wondering if there is a way to abort ampache's processing when this happens.  Or am I just overthinking this? I think I might be doing that.

**Comment by lachlan-00** [comment]
latest config is actually 43 so you're 44.

is the exception generated by spotify? it might not be possible to reduce that?

**Comment by wagnered** [comment]
Hold everything!

Please correct me if I'm wrong.

I just opened ampache.cfg.php.dist and found `$results['int_config_version'] = '40';`, which I believe is the correct number to change.  I did find `$results['user_ip_cardinality'] = 42;`.  The latter number agreed with the config file before I edited it.  The same situation in the develop branch.

So ampache's logic only checks for version numbers less than the one in init.php. Shouldn't it be checking for the exact version?

BTW is `user_ip_cardinality` a real thing?

**Comment by lachlan-00** [comment]
cardinality is days of IP record keeping so doesn't need to do anything with the config. i think it just got caught in a find and replace.

echonest is out now and spotify is in.
